### DISCUSSION TITLE
About api use,i can't get shorturl from my yourls used longurl?

### DISCUSSION THREAD

**Discussion by mangic** [YOURLS/YOURLS#3548] [Q&A] [answered]
https://shorturl.abc.com/yourls-api.php?signature=XXXXXX&action=url-stats&format=json&url=https://XXX.com/XXX/XXX/XXX
return:{"statusCode":404,"message":"Error: short URL not found"}
but
https://shorturl.abc.com/yourls-api.php?signature=XXXXXX&action=url-stats&format=json&shorturl=https://shorturl.abc.com/xyz
return:
{"statusCode":200,"message":"success","link":{"shorturl":"https:\/\/shorturl.abc.com\/xyz","url":"https://XXX.com/XXX/XXX/XXX","title":"......","timestamp":"......","ip":".......","clicks":"0"}}

what is going on???

**Comment by dgw** [comment, marked as answer]
Of course the long URL can't be found. This API method expects a `shorturl`:

https://github.com/YOURLS/YOURLS/blob/8f49f2156d4660dcd4a98d044ac23fc42ac519e4/includes/functions-api.php#L50-L59

Depending on how YOURLS is configured, there could be multiple keywords for the same link, so this action cannot accept a long URL to look up. However, there is a plugin that adds long-URL lookup: https://github.com/bryzgaloff/yourls-api-lookup-keywords-by-url-substr

**Reply by mangic** [reply]
thanks,it work
